Framework/PythonInterface/mantid/plots/__init__.py

```diff
@@ -239,11 +252,36 @@ def track_workspace_artist(self, workspace, artists, data_replace_cb=None,
                 def data_replace_cb(_, __):
                     logger.warning("Updating data on this plot type is not yet supported")
             artist_info = self.tracked_workspaces.setdefault(name, [])
+
             artist_info.append(_WorkspaceArtists(artists, data_replace_cb,
+                                                 is_normalized,
                                                  spec_num))
-
+            self.check_axes_distribution_consistency()
         return artists
 
+    @staticmethod
+    def _on_off_to_bool(on_or_off):
+        if on_or_off.lower() == 'on':
+            return True
+        elif on_or_off.lower() == 'off':
+            return False
+        raise ValueError("Argument must be 'On' or 'Off'!")
+
+    def check_axes_distribution_consistency(self):
+        """
+        Checks if new workspace to be plotted is consistent with current
+        workspaces on axes in regard to being plotted as distributions.
+        Displays a warning if not.
+        """
+        tracked_ws_distributions = []
+        for artists in self.tracked_workspaces.values():
+            for artist in artists:
+                tracked_ws_distributions.append(artist.is_normalized)
+        if len(tracked_ws_distributions) > 0:
```

Comment:
This is a slightly confusing if group - do I have it correct that we post a logger warning if there is at least one artist in the list of artists, and at least one of those artists, but not all, are normalized?

A clearer statement might be to set `num_normalized = sum(tracked_ws_distributions)` and then `if num_normalized > 0 and num_normalized < len(tracked_ws_distributions)`

Reply:
`if not all() and any()` returns `False` if all items in the list are true or all items in the list are false. So it returns `True` if there's a `True` and a `False` in the list.

For consistency all artists on a plot should be distributions or all artists should not be, so a warning should be displayed if the user is plotting distribution data over non-distribution (or vice-versa). A similar warning is shown in MantidPlot.

Your way could work with something like:
```python
num_normalized = sum(tracked_ws_distributions)
if num_normalized == 0 or ((num_normalized) > 0 and num_normalized < len(tracked_ws_distributions)):
```
if you think that's clearer?

Reply:
I think that is clearer. We can either use that, or add a comment to the existing clause explaining that it will return True if there are at least one True and one False. I prefer `num_normalized == 0....` but I don't mind either way